<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <include
    file="$(find gazebo_ros)/launch/empty_world.launch" />

  <group ns="/agrobot_arm_v2">
    
    <!-- Robot model --> 
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find agrobot_arm_v2)/urdf/agrobot_arm_v2.urdf'" />
 
    <arg name="x" default="0" />
    <arg name="y" default="0" />
    <arg name="z" default="0.0373" />

    <!-- Spawn the robot model --> 
    <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen"
          args="-urdf -param robot_description -model mrm -x $(arg x) -y $(arg y) -z $(arg z)" />

    <!-- Load controllers -->
    <rosparam command="load" file="$(find agrobot_arm_v2)/config/joints.yaml" />
 
    <node name="controller_spawner" pkg="controller_manager" type="spawner"
          respawn="false" output="screen" ns="/agrobot_arm_v2"
          args="--namespace=/agrobot_arm_v2
          joint_state_controller
          joint1_position_controller
          joint2_position_controller
          joint3_position_controller
          joint4_position_controller
          --timeout 60">
    </node>
  </group>

  <!-- Start robot controller simulation python file -->
  <node name="motor_controller_simulation" pkg="agrobot_arm_v2" type="motor_controller_simulation.py" output="screen"/>

</launch>
